lib/util/date.js

```diff
@@ -0,0 +1,261 @@
+'use strict'
+
+const IMF_DAYS = ['mon', 'tue', 'wed', 'thu', 'fri', 'sat', 'sun']
+const IMF_SPACES = [4, 7, 11, 16, 25]
+const IMF_MONTHS = ['jan', 'feb', 'mar', 'apr', 'may', 'jun', 'jul', 'aug', 'sep', 'oct', 'nov', 'dec']
+const IMF_COLONS = [19, 22]
+
+const ASCTIME_SPACES = [3, 7, 10, 19]
+
+const RFC850_DAYS = ['monday', 'tuesday', 'wednesday', 'thursday', 'friday', 'saturday', 'sunday']
+
+/**
+ * @see https://www.rfc-editor.org/rfc/rfc9110.html#name-date-time-formats
+ *
+ * @param {string} date
+ * @param {Date} [now]
+ * @returns {Date | undefined}
+ */
+function parseHttpDate (date, now) {
+  // Sun, 06 Nov 1994 08:49:37 GMT    ; IMF-fixdate
+  // Sun Nov  6 08:49:37 1994         ; ANSI C's asctime() format
+  // Sunday, 06-Nov-94 08:49:37 GMT   ; obsolete RFC 850 format
+
+  date = date.toLowerCase()
+
+  if (date[3] === ',') {
+    return parseImfDate(date)
+  } else if (date[3] === ' ') {
+    return parseAscTimeDate(date)
+  } else {
+    return parseRfc850Date(date, now)
+  }
```

Comment:
Maybe a switch here?
```suggestion
  switch (date[3]) {
    case ',': return parseImfDate(date)
    case ' ': return parseAscTimeDate(date)
    default: return parseRfc850Date(date, now)
  }
```